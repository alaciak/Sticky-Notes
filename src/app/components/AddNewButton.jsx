import React from 'react';
import { addNote } from '../actions/notesBoardActions';
import { connect } from 'react-redux';

class AddNewButton extends React.Component {

  handleOnClickAdd = event => {
    event.preventDefault();
    this.props.addNote()
  }

  render() {
    return (
      <button type='button' className='btn btn-primary' onClick={ this.handleOnClickAdd }>Add Note</button>
    );
  }
}


const mapStateToProps = (state) => {
  return {}
};

const mapDispatchToProps = (dispatch) => {
  return {
    addNote: () => {
      dispatch(addNote());
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(AddNewButton);
