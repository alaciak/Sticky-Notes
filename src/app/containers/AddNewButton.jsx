import React from 'react';
import { addNote } from '../actions/notesBoardActions';
import { connect } from 'react-redux';

class AddNewButton extends React.Component {

  handleOnClickAdd = event => {
    event.preventDefault();
    this.props.addNote();
  }

  render() {
    return (
      <button type='button' className='btn btn-primary btn-add' onClick={ this.handleOnClickAdd } >Add Note</button>
    );
  }
}

const mapStateToProps = (state) => {
  return {}
};

const mapDispatchToProps = (dispatch) => {
  return {
    addNote: (note) => {
      dispatch(addNote(note));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(AddNewButton);
