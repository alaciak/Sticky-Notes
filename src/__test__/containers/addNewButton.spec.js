import React from 'react';
import { AddNewButton } from '../../app/containers/AddNewButton.jsx';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('AddNewButton', () => {

  it('should render AddNewButton component', () => {

    const wrapper = shallow(<AddNewButton />);

    expect(toJson(wrapper)).toMatchSnapshot();

  });

});
