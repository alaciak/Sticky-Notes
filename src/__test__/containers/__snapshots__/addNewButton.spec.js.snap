// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewButton should render AddNewButton component 1`] = `
<div
  className="row"
>
  <button
    className="btn btn-primary btn-add"
    onClick={[Function]}
    type="button"
  >
    Add Note
  </button>
</div>
`;
